test/e2e/contract-test/contract.js

```diff
@@ -169,19 +171,24 @@ web3.currentProvider.enable().then(() => {
 
   ethereum.autoRefreshOnNetworkChange = false
 
-  const networkDiv = document.getElementById('network')
-  const chainIdDiv = document.getElementById('chainId')
-  const accountsDiv = document.getElementById('accounts')
-
-  ethereum.on('networkChanged', (networkId) => {
-    networkDiv.innerHTML = networkId
+  getAccountsButton.addEventListener('click', async function () {
+    const accounts = await ethereum.send({ method: 'eth_accounts' })
+    getAccountsResult.innerHTML = accounts[0] || 'Not able to get accounts'
   })
+})
 
-  ethereum.on('chainIdChanged', (chainId) => {
-    chainIdDiv.innerHTML = chainId
-  })
+const networkDiv = document.getElementById('networkDiv')
+const chainIdDiv = document.getElementById('chainIdDiv')
+const accountsDiv = document.getElementById('accountsDiv')
 
```

Comment:
These listeners have been moved outside the callback given to the `then` on the promise returned by enable.

This was only necessary for `'accountsChanged'`, in order to get the `ethereum-on.spec` e2e tests passing. It was done with the other listeners for consistency.

Previous to `LoginPerSite`, and in particular the inpage provider changes on LoginPerSite, the e2e tests pass without this change. Meanwhile, it does make sense to me that the listeners would be set up outside the callback.

I should review this with @rekmarks to make sure that there are no breaking changes with respect to when the `accountsChanged` listener needs to be set.

Reply:
- [ ] Review different listener behaviour after enable call with @rekmarks